app/controllers/witnesses_controller.rb

```diff
@@ -147,6 +147,34 @@ def assign
         @hosts = Host.where(city_id: get_city_ids_for_assignment, survivor_needed: true)
       end
 
+      if params[:floor]
+        if params[:floor][0] == '>'
+          above_floor = params[:floor][1..-1].to_i
+          @hosts = @hosts.where('floor > ?', above_floor)
+        else
+          floor = params[:floor].to_i
+          @hosts = @hosts.where(floor: floor)
+        end
+      end
+
+      if params[:elevator]
+        is_elevator = ActiveRecord::Type::Boolean.new.type_cast_from_user(params[:elevator])
+        @hosts = @hosts.where(elevator: is_elevator)
+      end
+
+      @hosts = @hosts.where(event_language: params[:event_language]) if params[:event_language]
+
+      if params[:available_day]
+        @hosts =
+          if params[:available_day] == 'available_day_other'
+            know_dates = (1..30).map { |day| available_day_to_available_date("available_day#{day}") }.compact
```

Comment:
if this is month days, there are 31 days on some months

Reply:
@aryehbeitz  there are currently 9 "available days". This is a number bigger than 9, just in case there will be more available days in the future. It doesn't have a special meaning.